lib/extensions/overlays.nix

```
@@ -0,0 +1,58 @@
+{ lib, delib, ... }:
+let
+  inherit (builtins) concatStringsSep;
+  inherit (delib) extension;
+  inherit (lib) elem mkIf optional;
+in
+extension {
+  name = "overlays";
+  description = "Simplified overlay configuration module";
+
+  config = final: prev: {
+    defaultTargets = [
+      "nixos"
+      "home"
+    ];
+    moduleNamePrefix = "overlay";
+  };
+
+  libExtension = config: final: _: {
+    overlayModule =
+      {
+        name,
+        overlay ? null,
+        overlays ? [ ],
+        targets ? config.defaultTargets,
+        withPrefix ? true,
+      }:
+      let
+        finalOverlays = overlays ++ (optional (overlay != null) overlay);
+
+        applyToNixOS = elem "nixos" targets;
+        applyToHomeManager = elem "home" targets;
+        applyToMacOS = elem "darwin" targets;
+      in
+      final.module {
+        name =
+          if withPrefix then
+            (concatStringsSep "." [
+              config.moduleNamePrefix
+              name
+            ])
+          else
+            name;
+
```

```suggestion

        options = delib.singleEnableOption true;

```
I think this option should be documented